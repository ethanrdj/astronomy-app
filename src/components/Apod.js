import React, { useEffect, useState } from "react";
import axios from "axios";

import Aos from "aos";
import "aos/dist/aos.css";

import "../styles/Apod2.css";

const Apod = () => {
  const [data, setData] = useState({});

  useEffect(() => {
    const fetchData = async () => {
      await axios
        .get(
          "https://api.nasa.gov/planetary/apod?api_key=oEknIXCWLLvYbvn5KqiSdC2wIhKRXjgN4aNiSLmF"
        )
        .then(({ data }) => {
          setData(data);
        });
    };
    fetchData();
  }, []);

  useEffect(() => {
    Aos.init({ duration: 1000, offset: 150 });
  }, []);

  return (
    <div className="Apod" data-testid="apod" id="apod">
      <h1 data-aos="fade-right">Astronomy Picture of The Day</h1>
      <h3 data-aos="fade-left">
        Each day NASA uploads an astronomical photo, here is todays..
      </h3>
      <div className="apod-data">
        <img
          data-aos="fade-right"
          className="apodImg"
          src={data.url}
          alt="apod"
        ></img>
        <div data-aos="fade-left" className="apod-info">
          <h2>
            {data.title} - <br />
            <span style={{ fontSize: "0.7em", fontStyle: "italic" }}>
              {data.date}
            </span>
          </h2>
          <p>{data.explanation}</p>
        </div>
      </div>
    </div>
  );
};

export default Apod;
