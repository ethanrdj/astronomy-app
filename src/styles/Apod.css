.Apod {
  width: 100vw;
  background-color: #e8e8e8;
  color: black;
}

.apod-data {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 0 auto;
}

.Apod h1 {
  text-align: center;
  font-family: "Oswald", sans-serif;
  text-transform: uppercase;
  font-weight: lighter;
}

.Apod h3 {
  font-family: "Oswald", sans-serif;
  font-weight: lighter;
  margin: 0;
  margin-bottom: 3%;
  width: 90%;
  text-align: center;
}

.Apod h2 {
  width: 94%;
  margin-top: 3%;
  text-align: center;
  font-size: 1.7em;
}

.Apod p {
  font-family: "Raleway", sans-serif;
  width: 80%;
  padding: 2%;
  line-height: 1.5em;
  font-weight: lighter;
  border: #008080 solid 3px;
  border-radius: 5px;
  box-shadow: 0 4px 8px 0 rgba(153, 153, 153, 0.425),
    0 6px 20px 0 rgba(56, 56, 56, 0.89);
}

.apodImg {
  width: 70%;
  border: 15px solid white;
  box-shadow: 0 4px 8px 0 rgba(153, 153, 153, 0.425),
    0 6px 20px 0 rgba(56, 56, 56, 0.89);
}
