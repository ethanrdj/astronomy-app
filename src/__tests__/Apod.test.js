import React from "react";
import { render } from "@testing-library/react";
import Apod from "../components/Apod";

describe("Apod", () => {
  const mockData = {
    title: "Mock Nasa Title",
    date: "25th December 2019",
    explanation: "Apod explanation mock",
  };

  it("renders correctly", () => {
    const { asFragment } = render(<Apod />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct text", () => {
    const { getByText, getByTestId } = render(<Apod data={mockData} />);

    expect(getByText("Astronomy Picture of The Day")).toBeInTheDocument();
    expect(
      getByText("Each day NASA uploads an astronomical photo, here is todays..")
    ).toBeTruthy();
    expect(getByTestId("apod")).toHaveClass("Apod");
  });
});
