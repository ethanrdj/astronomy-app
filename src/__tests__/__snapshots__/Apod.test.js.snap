// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Apod renders correctly 1`] = `
<DocumentFragment>
  <div
    class="Apod"
    data-testid="apod"
    id="apod"
  >
    <h1
      class="aos-init aos-animate"
      data-aos="fade-right"
    >
      Astronomy Picture of The Day
    </h1>
    <h3
      class="aos-init aos-animate"
      data-aos="fade-left"
    >
      Each day NASA uploads an astronomical photo, here is todays..
    </h3>
    <div
      class="apod-data"
    >
      <img
        alt="apod"
        class="apodImg aos-init aos-animate"
        data-aos="fade-right"
      />
      <div
        class="apod-info aos-init aos-animate"
        data-aos="fade-left"
      >
        <h2>
           - 
          <br />
          <span
            style="font-size: 0.7em; font-style: italic;"
          />
        </h2>
        <p />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
